build-system/tasks/storybook-react/index.js

```diff
@@ -0,0 +1,36 @@
+('use strict');
+
+const argv = require('minimist')(process.argv.slice(2));
+const {build: buildBinary} = require('../build');
+const {buildEnv, launchEnv} = require('../storybook');
+const {createCtrlcHandler} = require('../../common/ctrlcHandler');
+const {defaultTask: runAmpDevBuildServer} = require('../default-task');
+/**
+ * @return {Promise<void>}
+ */
+async function storybookReact() {
+  // React dist needs to be comparied before running
+  await buildBinary();
+  const {build = false, 'storybook_env': env = 'amp,preact'} = argv;
+  const envs = env.split(',');
+  if (!build && envs.includes('amp')) {
+    await runAmpDevBuildServer();
+  }
+  if (!build) {
+    createCtrlcHandler('storybook');
+  }
+  envs.map(build ? buildEnv : launchEnv);
```

Comment:
Suggestion since we're branching 3x on `build`, feel free to ignore if you find it less clear

```suggestion
  if (build) {
    envs.forEach(buildEnv);
    return;
  }
  
  if (envs.includes('amp')) {
    await runAmpDevBuildServer();
  }
  createCtrlcHandler('storybook');
  envs.forEach(launchEnv);
 
```

Reply:
I actually like this a lot more

Reply:
I like this a lot more. It's a good refactor.